### PR TITLE
Aq 66 random username generator

### PR DESCRIPTION
**Pull request by SlawCzech**
# Description

Random username generator, which returns a list of five usernames on default.
Separate readme file for explanations how it works.

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Manual testing.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings
- [X] New and existing unit tests pass locally with my changes
- [X] The related memme has been added

Ferocious work

### CODE REVIEW

**Review by martynawitkowska — CHANGES_REQUESTED** [review]

**Review by martynawitkowska — CHANGES_REQUESTED** [review]
And please lets test it out with some unit tests

File: quiz/backend/helpers/gen_username/generate.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            nouns.append(line.strip())
+
+    usernames = []
+    for _ in range(num_results):
+        adjective = random.choice(adjectives)
+        noun = random.choice(nouns)
+        num = str(random.randrange(100))
+        usernames.append(adjective + noun + num)
+
+    return usernames
```
Comment (ToJestKrzysio) [inline quiz/backend/helpers/gen_username/generate.py:20]: I've got two thoughts
1. List/Set comprehension for lines 14-20 if you want to stick to yielding a list

2. turning entire generate_username into generator
```
while True:
    adjective = random.choice(adjectives)
    noun = random.choice(nouns)
    num = str(random.randrange(100))
    yield f"{adjective}{noun}{num}"
```

3. Considering theres 330 and 340 words in each set, it gives us 112_000 unique nicknames adding numbers may be an overkill

**Review by ToJestKrzysio — CHANGES_REQUESTED** [review]
Almost missed the fact that there's no tests.

File: quiz/backend/helpers/gen_username/generate.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    dir_path = os.path.dirname(__file__)
+    adjectives, nouns = [], []
+    with open(os.path.join(dir_path, "data", "useradj.txt"), "r") as file_useradj:
+        for line in file_useradj:
+            adjectives.append(line.strip())
+    with open(os.path.join(dir_path, "data", "usernoun.txt"), "r") as file_usernoun:
+        for line in file_usernoun:
+            nouns.append(line.strip())
+
+    usernames = []
```
Comment (martynawitkowska) [inline quiz/backend/helpers/gen_username/generate.py:22]: add type hints for all variables

**Review by martynawitkowska — CHANGES_REQUESTED** [review]

File: quiz/backend/helpers/gen_username/generate.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import random
+
+
+def generate_username(num_results=5):
```
[the remark this answers is not in the document]
Comment (martynawitkowska) [reply quiz/backend/helpers/gen_username/generate.py]: This płace is a good start for python typing.

https://realpython.com/python-type-checking/

File: quiz/backend/helpers/gen_username/generate.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import os
+import random
+
+from typing import List
+
+def generate_username(num_results=5) -> List[str]:
```
Comment (martynawitkowska) [inline quiz/backend/helpers/gen_username/generate.py:6]: So what I had in mind was something like this article. Sorry for the misleading comment. 

https://realpython.com/python-type-checking/

**Review by MWasile — APPROVED** [review]
works good, in my opinion nickname should start with capital letter and the second word with a number should be separated by space

File: quiz/backend/helpers/gen_username/generate.py (hunk 1 above)
Comment (MWasile) [reply quiz/backend/helpers/gen_username/generate.py:30]: for me, making generator for sending grouped data to the frontend, like 5 items at once is unnecessary

File: quiz/backend/helpers/tests/test_generate.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from helpers.gen_username.generate import generate_username
+
+
+def test_if_usernames_unique():
+    usernames = generate_username()
+
+    assert len(set(usernames)) == len(usernames)
+
+
+def test_generate_nondefault_num_usernames():
```
Comment (martynawitkowska) [inline quiz/backend/helpers/tests/test_generate.py:12]: You still have tests that are using files from the project, and it means you are not running tests in an isolated environment. Sorry for being pesky here, but the idea of unit tests is to test a for e. g. function without any distractions.

File: quiz/backend/helpers/tests/test_generate.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import pytest
+
+from helpers.gen_username.generate import generate_username
+
+
+def test_usernames_file_read(monkeypatch, nouns_temp_file, adjectives_temp_file):
```
Comment (martynawitkowska) [inline quiz/backend/helpers/tests/test_generate.py:6]: I would rename this test to `test_username_generator_with_no_args` and then I would add `test_username_generator_with_args`.

**Review by martynawitkowska — APPROVED** [review]
Approving changes, but dont forget about squasing commits 😄

File: quiz/backend/helpers/gen_username/data/useradj.txt [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+elegant
+elite
+eloquent
+empathetic
+enchanting
+engaged
+enterprising
+enthusiastic
+entrepreneurial
+erotic
```
Comment (martynawitkowska) [inline quiz/backend/helpers/gen_username/data/useradj.txt:92]: do we allow words like this? I have it in black list

**Pull request merged**